pkg/user/tx_client.go

```diff
@@ -798,7 +818,15 @@ func (client *TxClient) checkAccountLoaded(ctx context.Context, account string)
 	if err != nil {
 		return fmt.Errorf("retrieving address from keyring: %w", err)
 	}
-	// FIXME: have a less trusting way of getting the account number and sequence
+
+	if !client.trustedMode {
+		// Non-trusted mode would implement additional verification here
+		// TODO: implement non-trusted account verification
+		return fmt.Errorf("non-trusted mode not yet implemented")
```

Comment:
This should return a more helpful error like:
```suggestion
		return fmt.Errorf("no info for account %s found", account)
```
You could also check this earlier on line 183